### PR TITLE
Only warn of rate-limits when using HF endpoint

### PR DESCRIPTION
**Pull request by HennerM**
I am trying the llm-vscode extension with llm-ls on a locally hosted endpoint (running a custom fine-tuned model), however the extension still gives a warning that I might get rate limited by HuggingFace.

Since inference doesn't run on a HuggingFace server this warning is not necessary.

### CODE REVIEW

**Review by McPatate — COMMENTED** [review]
Since we know have the adaptor setting, I'd rather check its value than doing so w/ the URL. Wdyt?

**Comment by HennerM** [comment]
> Since we know have the adaptor setting, I'd rather check its value than doing so w/ the URL. Wdyt?

Yes good idea, I changed this to just check the adopter value now

File: crates/llm-ls/src/adaptors.rs [hunk 1]
```diff
@@ -200,11 +200,11 @@ fn parse_openai_text(text: &str) -> Result<Vec<Generation>, jsonrpc::Error> {
...
-const TGI: &str = "tgi";
-const HUGGING_FACE: &str = "huggingface";
-const OLLAMA: &str = "ollama";
-const OPENAI: &str = "openai";
-const DEFAULT_ADAPTOR: &str = HUGGING_FACE;
+pub(crate) const TGI: &str = "tgi";
+pub(crate) const HUGGING_FACE: &str = "huggingface";
+pub(crate) const OLLAMA: &str = "ollama";
+pub(crate) const OPENAI: &str = "openai";
+pub(crate) const DEFAULT_ADAPTOR: &str = HUGGING_FACE;
```
Comment (McPatate) [inline crates/llm-ls/src/adaptors.rs:207]: This should be an enum 🤔

I'll open a separate issue

File: crates/llm-ls/src/main.rs [hunk 2]
```diff
@@ -26,6 +26,7 @@ mod language_id;
 const MAX_WARNING_REPEAT: Duration = Duration::from_secs(3_600);
 pub const NAME: &str = "llm-ls";
 pub const VERSION: &str = env!("CARGO_PKG_VERSION");
+const HUGGINGFACE_INFERENCE_HOSTNAME: &str = "api-inference.huggingface.co";
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:29]:
```suggestion
const HF_INFERENCE_API_HOSTNAME: &str = "api-inference.huggingface.co";
```

File: crates/llm-ls/src/main.rs [hunk 3]
```diff
@@ -613,7 +614,8 @@ impl Backend {
                 "received completion request for {}",
                 params.text_document_position.text_document.uri
             );
-            if params.api_token.is_none() {
+            let is_using_hf_inference = params.adaptor.as_ref().unwrap_or(&adaptors::DEFAULT_ADAPTOR.to_string()).as_str() == adaptors::HUGGING_FACE;
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:617]:
```suggestion
            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(&adaptors::DEFAULT_ADAPTOR.to_string()).as_str() == adaptors::HUGGING_FACE;
```

**Review by McPatate — CHANGES_REQUESTED** [review]
Small nits on naming, we should be good to go!

File: crates/llm-ls/src/main.rs [hunk 4]
```diff
@@ -613,7 +614,8 @@ impl Backend {
                 "received completion request for {}",
                 params.text_document_position.text_document.uri
             );
-            if params.api_token.is_none() {
+            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(&adaptors::DEFAULT_ADAPTOR.to_string()).as_str() == adaptors::HUGGING_FACE;
+            if params.api_token.is_none() && is_using_hf_inference {
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:618]:
```suggestion
            if params.api_token.is_none() && is_using_inference_api {
```
Should've added all suggestions sorry!

File: crates/llm-ls/src/main.rs [hunk 5]
```diff
@@ -584,7 +585,7 @@ fn build_url(model: &str) -> String {
     if model.starts_with("http://") || model.starts_with("https://") {
         model.to_owned()
     } else {
-        format!("https://api-inference.huggingface.co/models/{model}")
+        format!("https://{HUGGINGFACE_INFERENCE_HOSTNAME}/models/{model}")
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:588]:
```suggestion
        format!("https://{HF_INFERENCE_API_HOSTNAME}/models/{model}")
```

**Review by McPatate — CHANGES_REQUESTED** [review]

File: crates/llm-ls/src/main.rs [hunk 6]
```diff
@@ -613,7 +614,8 @@ impl Backend {
                 "received completion request for {}",
                 params.text_document_position.text_document.uri
             );
-            if params.api_token.is_none() {
+            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(&adaptors::DEFAULT_ADAPTOR.to_string()).as_str() == adaptors::HUGGING_FACE;
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:617]:
```suggestion
            let is_using_inference_api = params.adaptor.as_ref().unwrap_or_else(|| &adaptors::DEFAULT_ADAPTOR.to_owned()).as_str() == adaptors::HUGGING_FACE;
```

File: crates/llm-ls/src/main.rs [hunk 7]
```diff
@@ -613,7 +614,8 @@ impl Backend {
                 "received completion request for {}",
                 params.text_document_position.text_document.uri
             );
-            if params.api_token.is_none() {
+            let is_using_inference_api = params.adaptor.as_ref().unwrap_or_else(|| &adaptors::DEFAULT_ADAPTOR.to_owned()).as_str() == adaptors::HUGGING_FACE;
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:617]:
```suggestion
            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(adaptors::DEFAULT_ADAPTOR).as_str() == adaptors::HUGGING_FACE;
```

File: crates/llm-ls/src/main.rs [hunk 8]
```diff
@@ -613,7 +614,8 @@ impl Backend {
                 "received completion request for {}",
                 params.text_document_position.text_document.uri
             );
-            if params.api_token.is_none() {
+            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(adaptors::DEFAULT_ADAPTOR).as_str() == adaptors::HUGGING_FACE;
```
Comment (McPatate) [inline crates/llm-ls/src/main.rs:617]:
```suggestion
            let is_using_inference_api = params.adaptor.as_ref().unwrap_or(&adaptors::DEFAULT_ADAPTOR.to_owned()).as_str() == adaptors::HUGGING_FACE;
```

**Pull request merged**